include/seqan3/range/view/trim.hpp

```diff
@@ -25,113 +24,48 @@ namespace seqan3::detail
-        //!\brief Reference to the parent
-        trim_fn const & parent;
+        static_assert(QualityAlphabet<remove_cvref_t<threshold_t>> || std::Integral<threshold_t>,
+                      "The threshold must either be a quality alphabet or an integral type "
+                      "in which case it is compared with the underlying phred type.");
 
-        //!\brief The operator() that only takes the range as argument and forwards to the two-parameter operator().
-        template <typename irng_t>
-        auto operator()(irng_t && irange) const
-        {
-            return parent(std::forward<irng_t>(irange), threshold);
-        }
-    };
-
-    /*!\brief Range-less interface for use with the pipe notation.
-     * \tparam threshold_t Must be an integral type or satisfy the seqan3::QualityAlphabet.
-     * \param threshold The minimum quality given by a value of the range's value type.
-     *
-     * \details
-     *
-     * Binds to one of the other interfaces and forwards the threshold.
-     */
-    template <typename threshold_t>
-    delegate<threshold_t> operator()(threshold_t const threshold) const
-    //!\cond
-        requires std::is_integral_v<std::decay_t<threshold_t>> || QualityAlphabet<std::decay_t<threshold_t>>
-    //!\endcond
-    {
-        return delegate<threshold_t>{threshold, *this};
-        // this doesn't work here, see seqan3::detail::trim_fn::delegate.
-        //return std::bind(trim_fn(), std::placeholders::_1, threshold);
+        return adaptor_from_functor{*this, threshold};
     }
 
-    /*!\brief Pipe operator that enables view-typical use of pipe notation.
+    /*!\brief Trim based on minimum phred score.
      * \tparam irng_t The type of the range being processed. See seqan3::view::trim for requirements.
-     * \tparam threshold_t Must be of `irng_t`'s `value_type` or that `value_type`'s seqan3::underlying_phred_t.
-     * \param irange The range being processed as left argument of the pipe.
-     * \param bound_view The result of the single-argument operator() (interface with bound threshold parameter).
+     * \param irange The range being processed.
+     * \param threshold The minimum quality as a phred score [integral type].
      */
-    template <typename irng_t,
-              typename threshold_t>
-    //!\cond
-        requires std::ranges::InputRange<irng_t> && QualityAlphabet<value_type_t<std::decay_t<irng_t>>> &&
-                 (std::is_same_v<std::decay_t<threshold_t>,
-                                 std::decay_t<value_type_t<std::decay_t<irng_t>>>> ||
-                  std::is_convertible_v<std::decay_t<threshold_t>,
-                                        underlying_phred_t<std::decay_t<value_type_t<std::decay_t<irng_t>>>>>)
-    //!\endcond
-    friend auto operator|(irng_t && irange,
-                          seqan3::detail::trim_fn::delegate<threshold_t> const & bound_view)
+    template <std::ranges::InputRange irng_t, typename threshold_t>
```

Comment:
There is no constraint on the range but you are calling `to_phred()` on its values. I would add a constraint. 
Is `to_phred` defined for all Integral types?

Reply:
We said that we would enforce these constraints via static_assert now which is exactly what I am doing below.